src/fetchtastic/download/orchestrator.py

```
@@ -1864,67 +1845,166 @@ def _calculate_success_rate(self) -> float:
             for result in self.download_results
             if result.success and getattr(result, "was_skipped", False) is not True
         )
-        attempted = downloaded_count + len(self.failed_downloads)
+        failed_count = len(self._get_unique_failures())
+        attempted = downloaded_count + failed_count
         return (downloaded_count / attempted) * 100 if attempted > 0 else 100.0
 
-    def _count_artifact_downloads(self, artifact_type: str) -> int:
+    def _count_artifact_downloads(
```

Ruff (0.15.12)

[warning] Too many branches (15 > 12)

(PLR0912)